infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -677,8 +686,19 @@ private void updateReplicationManagerTopology(TopologyDescriptor newConfig) {
     private void processUpgrade(DiscoveryServiceEvent event) {
         if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE) {
             // TODO pankti: is this correct?
-            replicationManager.restart(event.getRemoteSiteInfo());
+            replicationManager.restart(event.getRemoteClusterInfo());
+        }
+    }
+
+    /**
+     * Enforce a snapshot full sync for all standbys if the current node is a leader node
+     */
+    private void processEnforceSnapshotSync(DiscoveryServiceEvent event) {
+        if (replicationManager == null || !isLeader.get()) {
+            return;
         }
+
+        replicationManager.enforceSnapshotSync(event);
```

Comment:
I think the logic here is not correct.
Assuming caller sends enforce request to all 3 nodes, and current node is non leader.
So processEnforceSnapshotSync is done here, but `public boolean forceSnapshotSync()` is still blocking until timeout. 

Maybe we can completeExceptionally the cf with an Illegal State Exception. So we will see ExecutionException and you already have the log.